docs: add new workflow documents (#1174)
ti-community-infra/tichi: docs/en/workflows/pr.md

# PR Workflow 

## Design Background 

We use prow's native lgtm + approve related plug-ins to drive the process, but we have added the optional configuration of multi-person lgtm on this basis. 

> ## Most of the PR collaboration process 

Reviewer: ```suggestion
## Main PR Collaboration Process 
```

# PR Workflow 

## Design Background 

We use prow's native lgtm + approve related plug-ins to drive the process, but we have added the optional configuration of multi-person lgtm on this basis. 

## Main PR Collaboration Process
- The **author** submits a PR
- Phase 0: Automation suggests **[reviewers][reviewer-role]** and **[approvers][approver-role]** for the PR

>   - Determine the set of OWNERS files nearest to the code being changed

Reviewer: better to add external link for OWNERS file

  - [prow](https://prow.tidb.net) ([@ti-chi-bot](https://github.com/apps/ti-chi-bot)) updates its
    comment in the PR to indicate which **approvers** still need to approve
  - Once all **approvers** (one from each of the previously identified OWNERS files) have approved,
    [prow](https://prow.tidb.net) ([@ti-chi-bot](https://github.com/apps/ti-chi-bot)) applies an
    `approved` label
- Phase 3: Automation merges the PR:
  - If all of the following are true:
    - All required labels are present (eg: `lgtm`, `approved`)
    - Any blocking labels are missing (eg: there is no `do-not-merge/hold`, `needs-rebase`)
  - And if any of the following are true:
    - there are no presubmit prow jobs configured for this repo
    - there are presubmit prow jobs configured for this repo, and they all pass after automatically
      being re-run one last time
  - Then the PR will automatically be merged

> Modified based on [kubernetes community review process](https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#code-review-using-owners-files).

## Recommended configuration items 

### It is recommended to use Squash mode to merge code

> In terms of merging, we still recommend using GitHub's Squash mode for merging, because this is the current tradition of the TiDB community. Everyone will create a large number of commits in the PR, and then automatically perform Squash through GitHub when merging. At present, our ti-community-merge is also designed to serve the Squash mode. **If you do not use the Squash mode, then you need to be responsible for rebase or squash PR in the PR, which will invalidate our function of storing and submitting the hash (details See Q&A), eventually causing status/can-merge to automatically cancel** because there are new commits. So we strongly recommend that you use Squash mode for collaboration. 

Reviewer: ```suggestion
In terms of merging, we still recommend using GitHub's Squash mode for merging, because this is the current tradition of the TiDB community. Everyone will create a large number of commits in the PR, and then automatically perform Squash through GitHub when merging. At present, our ti-community-merge is also designed to serve the Squash mode. **If you do not use the Squash mode, then you need to be responsible for rebase or squash commits in the PR, which will invalidate our function of storing and submitting the hash (details See Q&A), eventually causing status/can-merge to automatically cancel** because there are new commits. So we strongly recommend that you use Squash mode for collaboration. 
```

> ### If the CI task of the repository is triggered by prow, you need to close the Require branches to be up to date before merging branch protection option. 

Reviewer: ```suggestion
### If the CI task of the repository is triggered by prow, you need to trun off the "Require branches to be up to date before merging branch protection" option. 
```
